app/jobs/qualification/expiration_mailer_job.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+class Qualification::ExpirationMailerJob < RecurringJob
+  run_every 1.day
+
+  private
+
+  def perform_internal
+    send_reminder(..1.day.from_now, Qualification::ExpirationMailer::REMINDER_TODAY)
+
+    return unless Time.zone.today == Time.zone.today.end_of_year
+
+    send_reminder(1.year.from_now.all_year, Qualification::ExpirationMailer::REMINDER_1_YEAR)
+    send_reminder(2.years.from_now.all_year, Qualification::ExpirationMailer::REMINDER_2_YEARS)
+  end
+
+  def send_reminder(finish_at, content_key)
+    Qualification.where(finish_at: finish_at).each do |qualification|
+      return unless qualification.person.qualifications.order(:finish_at).first == qualification
```

Comment:
dies würde ich auch in eine eigene Methode extrahieren oder sogar in eine eigene Domain Klasse.
wir versuchen die spezifische Logik möglichst gering zu halten in den Jobs selber.

sowas wie `Events::Qualifications::Expiring`

Reply:
dieses query müsste man wohl noch etwas optimieren. 
das gibt n+1 queries so.
ich finde es ist auch nicht so leserlich ... willst du da noch etwas optimieren?